### ISSUE TITLE
"this" as optional second argument in functions

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#5205]
## tl;dr

A user asked in a [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1720452417037999?thread_ts=1719937758.651189&cid=CTSMAK6G7):

> I noticed that the `grep` function takes `this` as the 2nd argument by default, and [...] most(?) / "all other"(?) functions don't. But now that I see they _could_, why don't they?

## Details

At the time this issue is being opened, Zed is at commit 343ac63.

In chatting with @mccanne about why the `grep` function currently behaves this way, he pointed out "that's how grep works" (i.e., at the Unix command line, `grep` just works off unnamed incoming lines of input). That said, the user's observation seems valid since we've already had some reports of user confusion between functions and operators, and seen through the eyes of a new user, I can see how this behavior of `grep` feels operator-like and could add to that confusion.

I'm not certain of the ideal way we might address this, but for now I'll link it to #4916 since that seems to be where we've been accumulating topics like this for later consideration.

**Comment by philrz** [comment]
This was made consistent via the changes in linked PR #6115.

Previously (such as at commit 886e33a right before the changes merged from #6115) the second argument to `grep` was optional and would default to `this` when not provided.

```
$ super -version
Version: 886e33aac

$ echo '"foobarbaz"' | super -c "grep('bar')" -
"foobarbaz"

$ echo '"foobarbaz"' | super -c "grep('bar', this)" -
"foobarbaz"
```

Whereas at commit 687d649 and newer, now the second argument is required.

```
$ super -version
Version: 687d6490f

$ echo '"foobarbaz"' | super -c "grep('bar')" -
too few arguments at line 1, column 1:
grep('bar')
~~~~~~~~~~~

$ echo '"foobarbaz"' | super -c "grep('bar', this)" -
"foobarbaz"
```

Docs updates that reflect this are coming as part of a wider docs revision.

We plan to be consistent with this pattern across functions going forward, so, closing this issue.
